src/utils/scraping.py

```diff
@@ -1,74 +1,101 @@
-    retieve the AMD database from out github repository
+    retieve the database (either intel or amd) from out github repository
 
     :param fname: name of the json file to be saved
     :return: dictionary containing the AMD database
     """
     # url of the json file in our github repository
     # todo: update according to branch 
-    url = 'https://raw.githubusercontent.com/MattiaSilvestri/energy-monitor/cpu_data/data/tableExport.json'
+    url = f'https://raw.githubusercontent.com/MattiaSilvestri/energy-monitor/main/data/database_{cpu_brand}.json'
     # get the json file in the url using requests and parse it as json format
     data = requests.get(url).json()
     # save it in the data folder
     with open(fname, 'w') as f:
         json.dump(data, f)
-    return data
\ No newline at end of file
+
+    return data
+
+
+def _create_intel_database():
+    """
+    Create the database of Intel CPUs and their TDPs, scraping information from the intel website.
+
+    """
+    def _isintel(tag):
+        return tag.has_attr('data-value') and 'Intel' in tag['data-value']
+    
+    # initialize the dictionary
+    intel_database = dict()
+
+    # get the list of processors families from the intel website
+    search_url = "https://www.intel.com/content/www/us/en/products/details/processors.html"
+    soup = BeautifulSoup(requests.get(search_url).content, 'html.parser')
+    # find the title class
+    group_list = soup.find_all("div", attrs={"class": "group-title has-name"})
+    # iterate over processor groups
+    for group in group_list:
+        # get href, insert 'products' before .html and transform href to url
+        group_url = 'https://www.intel.com' + group.find("a")['href'].replace('.html', '/products.html')
+        soup = BeautifulSoup(requests.get(group_url).content, 'html.parser')
+        # stop the loop for a random time between 0 and 1 seconds to avoid being blocked
+        time.sleep(np.random.rand())
+        # isolate the list of processors by looking for the data-value that contains 'Intel'
+        processor_list = soup.find_all(_isintel)
+        # if it is empty, skip
+        if not processor_list:
+            continue
+        # iterate over processors
+        for processor in processor_list:
+            # get href and transform to url
+            processor_url = 'https://www.intel.com' +  processor.find("a")['href']
+            # request the processor page
+            soup = BeautifulSoup(requests.get(processor_url).content, 'html.parser')
+            # stop the loop for a random time between 0 and 1 seconds to avoid being blocked
+            time.sleep(np.random.rand())
+            # get the tdp
+            processor_specs = soup.find_all("div", attrs={"class": "row tech-section-row"})
+            # iterate over specs
+            for spec in processor_specs:
+                # check if tech-label contains 'TDP'
+                if spec.find("div", attrs={"class": "tech-label"}) is not None:
```

Comment:
Isn't it more readible to just put `if spec.find("div", attrs={"class": "tech-label"}):` without the `is not None` part?